posts/2024-03-28-open-liberty-with-langchain4j-example.adoc

```diff
@@ -0,0 +1,341 @@
+* Create a new access token with "read" role
+
+To access the repository remotely you'll need to install https://git-scm.com/book/en/v2/Getting-Started-Installing-Git[Git] if you haven't already. You can clone the `langchain4j-examples` GitHub repository to your local machine by running:
+
+[source]
+----
+git clone https://github.com/langchain4j/langchain4j-examples.git
+----
+
+[#env_setup]
+=== Environment Set Up
+
+To run the application, navigate to the `jakartaee-microprofile-example` directory:
+
+[source]
+----
+cd langchain4j-examples/jakartaee-microprofile-example
+----
+
+and set the environment variables:
+
+[source]
+----
+export JAVA_HOME=<your Java 17 home path>
+export HUGGING_FACE_API_KEY=<your Hugging Face read token>
+----
+
+[#start_app]
+=== Start the application
+
+To start the application, use the provided Maven wrapper to run the https://openliberty.io/docs/latest/development-mode.html[Liberty dev mode]:
+
+
+[source]
+----
+./mvnw liberty:dev
+----
+
+After you see the following message, the application is ready:
+
+----
+************************************************************************
+*    Liberty is running in dev mode.
+*        Automatic generation of features: [ Off ]
+*        h - see the help menu for available actions, type 'h' and press Enter.
+*        q - stop the server and quit dev mode, press Ctrl-C or type 'q' and press Enter.
+*
+*    Liberty server port information:
+*        Liberty server HTTP port: [ 9080 ]
+*        Liberty server HTTPS port: [ 9443 ]
+*        Liberty debug port: [ 7777 ]
+************************************************************************
+----
+
+To ensure the application has started successfully, you can run the tests by pressing the `enter/return` key from the command-line session. If the tests pass, you should see a similar output to the following example:
+
+----
+[INFO] -------------------------------------------------------
+[INFO]  T E S T S
+[INFO] -------------------------------------------------------
+[INFO] Running it.dev.langchan4j.example.ChatServiceIT
+[INFO] ...
+[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.439 s...
+[INFO] ...
+[INFO] Running it.dev.langchan4j.example.ModelResourceIT
+[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.733 s...
+[INFO]
+[INFO] Results:
+[INFO]
+[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0
+----
+
+[#access_app]
+=== Access the application
+
+Once the application is running, you can access it through a browser of your choice at http://localhost:9080/ and start experimenting with it.
+
+image::/img/blog/langchain4j-example-chat-room.png[Chat Room of LangChain4j Jakarta EE and MicroProfile example,width=70%,align="center"]
+
+You can type in any text that you want to chat with the AI agent. Here are some suggested messages:
+
+* `What is MicroProfile?`
+* `Which top 10 companies contribute MicroProfile?`
+* `any documentation?`
```

Comment:
```suggestion
* `Any documentation?`
```

Reply:
this seems like too vague of a prompt to be effective but if it has been tested in context we can go with it